File: PHP/Kode/Ponsel.php
<?php
require_once 'ProdukElektronik.php';

// Class turunan dari ProdukElektronik, khusus ponsel
class Ponsel extends ProdukElektronik {
    // Atribut private
    private $ukuranLayar;
    private $kapasitasPenyimpanan;
    private $ram;

    // Konstruktor
    public function __construct($nama = "", $merek = "", $harga = 0, $stok = 0, $foto_produk = "no-image.png", $konsumsiDaya = 0, $voltase = 0, $garansi = 0, $ukuranLayar = 0.0, $kapasitasPenyimpanan = 0, $ram = 0) {
        // Memanggil konstruktor induk
        parent::__construct($nama, $merek, $harga, $stok, $foto_produk, $konsumsiDaya, $voltase, $garansi);
        $this->ukuranLayar = $ukuranLayar;
        $this->kapasitasPenyimpanan = $kapasitasPenyimpanan;
        $this->ram = $ram;
    }

    // Setter dan Getter untuk setiap atribut
    public function setUkuranLayar($ukuranLayar) {
        $this->ukuranLayar = $ukuranLayar;
    }

    public function getUkuranLayar() {
        return $this->ukuranLayar;
    }

    public function setKapasitasPenyimpanan($kapasitasPenyimpanan) {
        $this->kapasitasPenyimpanan = $kapasitasPenyimpanan;
    }

    public function getKapasitasPenyimpanan() {
        return $this->kapasitasPenyimpanan;
    }

    public function setRam($ram) {
        $this->ram = $ram;
    }

    public function getRam() {
        return $this->ram;
    }
}

File: CPP/Kode/Ponsel.cpp
#pragma once

#include "ProdukElektronik.cpp"

class Ponsel : public ProdukElektronik { // Class turunan dari ProdukElektronik, khusus ponsel
    private: // Atribut private
        double ukuranLayar;
        int kapasitasPenyimpanan;
        int ram;

    public: // Method public
        Ponsel() { // Konstruktor default
        }

        Ponsel(string nama, string merek, int harga, int stok, int konsumsiDaya, int voltase, int garansi,
            double ukuranLayar, int kapasitasPenyimpanan, int ram) 
            : ProdukElektronik(nama, merek, harga, stok, konsumsiDaya, voltase, garansi) { // Memanggil konstruktor induk
            this->ukuranLayar = ukuranLayar;
            this->kapasitasPenyimpanan = kapasitasPenyimpanan;
            this->ram = ram;
        }

        // Setter dan Getter untuk setiap atribut

        void setUkuranLayar(double ukuranLayar) {
            this->ukuranLayar = ukuranLayar;
        }
        double getUkuranLayar() {
            return this->ukuranLayar;
        }

        void setKapasitasPenyimpanan(int kapasitasPenyimpanan) {
            this->kapasitasPenyimpanan = kapasitasPenyimpanan;
        }
        int getKapasitasPenyimpanan() {
            return this->kapasitasPenyimpanan;
        }

        void setRam(int ram) {
            this->ram = ram;
        }
        int getRam() {
            return this->ram;
        }

        ~Ponsel() { // Destruktor
        }
};